In [1]:
#Importar las librerias necesarias
import pandas as pd
import numpy as np

In [2]:
# Cargar el dataset
data=pd.read_csv("DataComplementaria.csv")

C:\Users\Usuario\AppData\Local\Temp\ipykernel_14112\3904205643.py:2: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  data=pd.read_csv("DataComplementaria.csv")


In [3]:
# Explorar el dataset
# Visualizar las primeras filas del DataFrame
data.head(3)

AIDS Report Number Local Event Date Event City Event State   Event Airport   
0                                NaN        NaN         NaN             NaN  \
1    19780101000019I        01-JAN-78   WAHPETON          ND    BRECKENRIDGE   
2    19780101000029I        01-JAN-78  FAIRBANKS          AK  FAIRBANKS INTL   

  Event Type Aircraft Damage           Flight Phase Aircraft Make   
0        NaN             NaN                    NaN           NaN  \
1   INCIDENT           MINOR  ROLL-OUT (FIXED WING)        CESSNA   
2   INCIDENT           MINOR  ROLL-OUT (FIXED WING)         PIPER   

  Aircraft Model  ... Total Injuries Aircraft Engine Make   
0            NaN  ...            NaN                  NaN  \
1            182  ...            0.0                  NaN   
2           PA18  ...            0.0                  NaN   

  Aircraft Engine Model Engine Group Code Nbr of Engines PIC Certificate Type   
0                   NaN               NaN            NaN                  NaN  \
1                   NaN               NaN            1.0        PRIVATE PILOT   
2                   NaN               NaN            1.0              STUDENT   

   PIC Flight Time Total Hrs  PIC Flight Time Total Make-Model        .1  
0                        NaN                               NaN  NaN  NaN  
1                      245.0                             136.0  0.0  0.0  
2                      200.0                               2.0  0.0  0.0  

[3 rows x 27 columns]

In [4]:
# Obtener información del dataset
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 27 columns):
 #   Column                            Non-Null Count   Dtype  
---  ------                            --------------   -----  
 0   AIDS Report Number                100000 non-null  object 
 1   Local Event Date                  99999 non-null   object 
 2   Event City                        98515 non-null   object 
 3   Event State                       99233 non-null   object 
 4   Event Airport                     81401 non-null   object 
 5   Event Type                        99999 non-null   object 
 6   Aircraft Damage                   71198 non-null   object 
 7   Flight Phase                      99757 non-null   object 
 8   Aircraft Make                     97441 non-null   object 
 9   Aircraft Model                    96928 non-null   object 
 10  Aircraft Series                   96927 non-null   object 
 11  Operator                          27412 non-null   ob

In [5]:
# Verificar duplicados
data.duplicated().value_counts()

False    100000
Name: count, dtype: int64

In [7]:
data.columns

Index(['AIDS Report Number', 'Local Event Date', 'Event City', 'Event State',
       'Event Airport', 'Event Type', 'Aircraft Damage', 'Flight Phase',
       'Aircraft Make', 'Aircraft Model', 'Aircraft Series', 'Operator',
       'Primary Flight Type', 'Flight Conduct Code', 'Flight Plan Filed Code',
       'Aircraft Registration Nbr', 'Total Fatalities', 'Total Injuries',
       'Aircraft Engine Make', 'Aircraft Engine Model', 'Engine Group Code',
       'Nbr of Engines', 'PIC Certificate Type', 'PIC Flight Time Total Hrs',
       'PIC Flight Time Total Make-Model', ' ', ' .1'],
      dtype='object')

In [8]:
# Seleccionar las columnas para el análisis univariado
columnas = ['AIDS Report Number', 'Local Event Date', 'Event City', 'Event State',
       'Event Airport', 'Event Type', 'Aircraft Damage', 'Flight Phase',
       'Aircraft Make', 'Aircraft Model', 'Aircraft Series', 'Operator',
       'Primary Flight Type', 'Flight Conduct Code', 'Flight Plan Filed Code',
       'Aircraft Registration Nbr', 'Total Fatalities', 'Total Injuries',
       'Aircraft Engine Make', 'Aircraft Engine Model', 'Engine Group Code',
       'Nbr of Engines', 'PIC Certificate Type', 'PIC Flight Time Total Hrs',
       'PIC Flight Time Total Make-Model', ' ', ' .1']

# Obtener el resumen estadístico de las columnas numéricas
resumen_estadistico = data[columnas].describe()

# Calcular la moda de las columnas categóricas
moda = data[columnas].mode()

# Calcular la tabla de frecuencia para las columnas categóricas
tabla_frecuencia = pd.DataFrame()
for col in columnas:
    if data[col].dtype == 'object':  # Considerar solo las columnas categóricas
        tabla_frecuencia[col] = data[col].value_counts()

# Mostrar los resultados
print("Resumen estadístico:")
print(resumen_estadistico)
print("\nModa:")
print(moda)
print("\nTabla de frecuencia:")
print(tabla_frecuencia)

Resumen estadístico:
       Total Fatalities  Total Injuries  Nbr of Engines   
count      99999.000000    99999.000000    93004.000000  \
mean           0.008220        0.050011        1.478539   
std            0.096502        0.659907        0.690930   
min            0.000000        0.000000        0.000000   
25%            0.000000        0.000000        1.000000   
50%            0.000000        0.000000        1.000000   
75%            0.000000        0.000000        2.000000   
max            5.000000       56.000000        4.000000   

       PIC Flight Time Total Hrs  PIC Flight Time Total Make-Model   
count               79806.000000                      85053.000000  \
mean                 3838.417913                        706.559287   
std                  5508.268646                       1624.157166   
min                     0.000000                          0.000000   
25%                   366.000000                         30.000000   
50%                  1540.0

In [9]:
# Seleccionar las columnas numéricas de interés
columnas_numericas = ['Total Fatalities', 'Total Injuries','Nbr of Engines']

# Crear un nuevo DataFrame con las columnas numéricas
data_numericas = data[columnas_numericas]

# Convertir las columnas a tipo numérico
data_numericas = data_numericas.astype(float)

# Calcular la matriz de correlación para las columnas numéricas
correlation_matrix = data_numericas.corr()

# Mostrar la matriz de correlación
print("Matriz de correlación:")
print(correlation_matrix)

Matriz de correlación:
                  Total Fatalities  Total Injuries  Nbr of Engines
Total Fatalities          1.000000        0.001710       -0.014291
Total Injuries            0.001710        1.000000        0.052428
Nbr of Engines           -0.014291        0.052428        1.000000


## CONCLUSIONES

### Según la información proporcionada por el método `info()` del objeto DataFrame, podemos extraer algunas conclusiones sobre los datos:

1. Tamaño de los datos: El DataFrame contiene 100,000 filas y 27 columnas.

2. Valores faltantes: Algunas columnas tienen valores faltantes. Las columnas con la cantidad más significativa de valores faltantes incluyen "Operator" (72,588 valores faltantes), "Engine Group Code" (69,898 valores faltantes), "Aircraft Engine Make" (63,606 valores faltantes), "Aircraft Engine Model" (63,602 valores faltantes), "PIC Flight Time Total Hrs" (20,194 valores faltantes), y "PIC Flight Time Total Make-Model" (14,947 valores faltantes).

3. Tipos de datos: El DataFrame contiene tanto columnas numéricas como columnas de tipo objeto. Las columnas numéricas incluyen valores de tipo float64, mientras que las columnas de tipo objeto pueden contener cadenas de texto u otros tipos de datos no numéricos.

4. Distribución de datos: Algunas columnas tienen una alta cantidad de valores faltantes, lo cual puede afectar el análisis de esos datos en particular. Además, se observa que algunas columnas relacionadas con detalles específicos del evento, como "Local Event Date", "Event City", "Event State" y "Event Airport", también tienen valores faltantes. Esto puede dificultar el análisis geográfico o temporal de los eventos.

En resumen, el DataFrame contiene datos de eventos relacionados con accidentes aéreos, con información variada sobre cada evento y detalles asociados, como daños a la aeronave, fase del vuelo, fabricante de la aeronave, número de motores, entre otros. Sin embargo, se debe tener en cuenta la presencia de valores faltantes en varias columnas, lo cual puede requerir manejo adicional durante el análisis de los datos.

### Basado en el resumen estadístico y la tabla de frecuencia proporcionados, podemos extraer las siguientes conclusiones:

1. El conjunto de datos contiene 100,000 registros y 27 columnas que representan diferentes atributos de informes de eventos relacionados con aviones.
2. Algunas columnas tienen valores faltantes, lo que indica que no todos los campos están completos en todos los registros. Las columnas con la mayoría de los valores faltantes son "Operator" y "Engine Group Code".
3. Algunas columnas tienen datos categóricos, como "Event City", "Event State", "Event Airport" y "Aircraft Make", mientras que otras tienen datos numéricos, como "Total Fatalities" y "Total Injuries".
4. En promedio, hubo muy pocas fatalidades y lesiones reportadas en los eventos relacionados. La media de "Total Fatalities" es aproximadamente 0.008 y la media de "Total Injuries" es aproximadamente 0.05.
5. La mayoría de los eventos involucraron aviones con 1 o 2 motores, con un promedio de aproximadamente 1.48 motores.
6. La columna "PIC Flight Time Total Hrs" muestra que el tiempo de vuelo promedio de los pilotos al mando es de aproximadamente 3,838 horas.
7. La columna "PIC Flight Time Total Make-Model" muestra que el tiempo de vuelo promedio en un modelo de avión específico es de aproximadamente 706 horas.
8. La columna ".1" tiene una media de 58.53 y una desviación estándar de 354.26. Dado que no se proporciona un nombre para esta columna, es difícil inferir su significado sin más información.

### Basándonos en la matriz de correlación, podemos obtener las siguientes conclusiones:

1. Hay una correlación muy débil entre "Total Fatalities" y "Total Injuries", con un coeficiente de correlación de        0.001710. Esto indica que no hay una relación significativa entre el número de fatalidades y el número de lesiones en los eventos analizados.

2. Existe una correlación débil positiva entre "Total Injuries" y "Nbr of Engines", con un coeficiente de correlación de 0.052428. Esto sugiere que a medida que aumenta el número de motores de un avión involucrado en un evento, hay una ligera tendencia al aumento de las lesiones reportadas. Sin embargo, esta correlación es bastante baja y no implica una relación causal directa.

3. No se observa una correlación significativa entre "Total Fatalities" y "Nbr of Engines", con un coeficiente de correlación de -0.014291. Esto indica que no hay una relación clara entre el número de fatalidades y el número de motores de un avión involucrado en los eventos analizados.

Es importante tener en cuenta que estas conclusiones se basan únicamente en los coeficientes de correlación y no tienen en cuenta otros factores o variables que pueden influir en los resultados. Además, la matriz de correlación solo muestra las relaciones lineales entre las variables y no captura relaciones no lineales o efectos causales directos. Para un análisis más completo, se recomienda realizar análisis adicionales y considerar más variables relevantes.